src/jdk.jfr/share/classes/jdk/jfr/internal/instrument/JDKEvents.java

```diff
@@ -300,4 +308,16 @@ private static void emitDirectBufferStatistics() {
         DirectBufferStatisticsEvent e = new DirectBufferStatisticsEvent();
         e.commit();
     }
+
+    private static void emitInitialSecurityProperties() {
+        Properties p = SharedSecrets.getJavaSecurityAccess().getInitialProperties();
+        if (p != null) {
+            for (Map.Entry<Object, Object> entry : p.entrySet()) {
+                InitialSecurityPropertyEvent e = new InitialSecurityPropertyEvent();
+                e.key = (String) entry.getKey();
+                e.value = (String) entry.getValue();
```

Comment:
To avoid any (odd/unexpected) `ClassCastException` here, perhaps this loop can be changed to something like:
```
for (Set<String> name : p.stringPropertyNames()) {
    InitialSecurityPropertyEvent e = new InitialSecurityPropertyEvent();
    e.key = name;
    e.value = p.getProperty(name);
```
Since this code anyway wants to deal with string key/values, this wouldn't introduce any functional change and yet at the same time prevent any unexpected/theoretical `ClassCastException`s

Reply:
nice - wasn't aware of that method in Properties. Code updated.